contracts/style.md

```diff
@@ -0,0 +1,215 @@
+
+### Delineate Unaudited Code
+
+- In a large repo it is worthwhile to keep code that has not yet been audited separate from the code that has been audited. This allows you to easily keep track of which files need to be reviewed.
+    - E.g. we keep unaudited code in a directory named `dev`. Only once it has been audited we move the audited files out of `dev` and only then is it considered safe to deploy.
+
+## Variables
+
+### Visibility
+
+- All contract variables should be private. Getters should be explicitly written and documented when you want to expose a variable publicly. Whether a getter function reads from storage, a constant, or calculates a value from somewhere else, that’s all implementation details that should not be exposed to the consumer by casing or other conventions.
+
+Examples:
+
+Good:
+
+```javascript
+uint256 private s_myVar;
+
+function getMyVar() external view returns(uint256){
+    return s_myVar;
+}
+```
+
+Bad:
+
+```javascript
+uint256 public s_myVar;
+```
+
+### Naming and Casing
+
+- Function arguments are named like this: `argumentName`. No leading or trailing underscores necessary.
+- Storage variables prefixed with an `s_` to make it clear that they live in storage and are expensive to read and write: `s_variableName`. They should always be private, and you should write explicit getters if you want to expose a storage variable.
+- Immutable variables should be prefixed with an `i_` to make it clear that they are immutable. E.g. `i_decimalPlaces`. They should always be private, and you should write explicit getters if you want to expose an immutable variable.
+- Internal/private constants should be all caps with underscores: `FOO_BAR`. Like other contract variables, constants should not be public. Create getter methods if you want to publicly expose constants.
+- Explicitly declare variable size: `uint256` not just `uint`. In addition to being explicit, it matches the naming used to calculate function selectors.
+
+Examples:
+
+Good:
+
+```javascript
+uint256 private s_myVar;
+uint256 private immutable i_myImmutVar;
+uint256 private constant MY_CONST_VAR;
+
+function multiplyMyVar(uint256 multiplier) external view returns(uint256){
+    return multiplier * s_myVar;
+}
+```
+
+Bad:
+
+```javascript
+uint private s_myVar;
+uint256 private immutable myImmutVar;
+uint256 private constant s_myConstVar;
+
+function multiplyMyVar_(uint _multiplier) external view returns(uint256){
+    return _mutliplier * s_myVar;
+}
+```
+
+### Types
+
+- If you are storing an address and know/expect it to be of a type(or interface), make the variable that type. This more clearly documents the behavior of this variable than the `address` type and often leads to less casting code whenever the address is used.
+
+Examples:
+
+Good:
+
+```javascript
+import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+// .
+// .
+// .
+AggregatorV3Interface private s_priceFeed;
```

Comment:
Using the `...Interface` naming structure here, but in the naming section, we recommend `ISomeContract` instead of `SomeContractInterface`.

Reply:
I've added a comment in to clarify old naming pattern and how we plan to switch to the more industry standard pattern in v0.9. Since we don't know when that is, I think it's best to leave it as is for now.